src/code.jl

```diff
@@ -1245,4 +1245,12 @@ function apply_optimization_rules(expr, state::Code.CSEState, rules)
     return nothing
 end
 
+function search_variables!(buf, expr::Let)
+    rhs_buf = empty(buf)
+    lhs_buf = empty(buf)
+    search_variables!.(Ref(rhs_buf), rhs.(expr.pairs))
+    search_variables!.(Ref(lhs_buf), lhs.(expr.pairs))
+    union!(buf, setdiff!(rhs_buf, lhs_buf))
```

Comment:
only issue i have with this is that the return semantics aren't well defined. We aren't guaranteed to have the `rhs_buf` populated returned as opposed to `nothing`

Reply:
Oh, I see. You're referring to return semantics of `setdiff!` - I assumed you were talking about `search_variables!` since this is a separate comment. Yeah, if you want this can be phrased as
```julia
setdiff!(rhs_buf, lhs_buf)
union!(buf, rhs_buf)
```